app/views/base.py

```diff
@@ -119,37 +133,23 @@ def build_absolute_path(self, path=None):
         use.
 
         Args:
-            path (str, optional): A path (may include a query string), e.g.,
-                'abc?x=y'. Paths beginning with a slash will be interpreted as
-                being relative to the Django app's directory, while paths
-                without a leading slash will be considered relative to the
-                current request's path. If the path argument is not specified,
-                None, or the empty string, the current request's path will be
-                used.
+            path (str, optional): A path beginning with a slash (may include a
+                query string), e.g., '/abc?x=y'. If the path argument is not
+                specified or is None, the current request's path will be used.
 
         Returns:
             str: An absolute path, including the sitewide OPTIONAL_PATH_PREFIX
             if it was used with the original request (e.g.,
             '/personfinder/abc?x=y'). Does not preserve query parameters from
             the original request.
         """
-        if not path:
+        if path is None:
             # request.path will already include the path prefix if it's being
             # used.
             return self.request.path
-        elif path[0] == '/':
-            # We're dealing with an absolute path, so add the path prefix if
-            # necessary.
-            return ('/%s%s' % (site_settings.OPTIONAL_PATH_PREFIX, path)
-                    if self._request_is_for_prefixed_path()
-                    else path)
-        else:
-            # We're dealing with a relative path, so just append it to the
-            # current request's path (which will already include the path prefix
-            # if it's being used).
-            return ('%s%s' % (self.request.path, path)
-                    if self.request.path[-1] == '/'
-                    else '%s/%s' % (self.request.path, path))
+        return ('/%s%s' % (site_settings.OPTIONAL_PATH_PREFIX, path)
```

Comment:
Can you add an assertion that it starts with "/"?

Reply:
Done.